pkg/types/gcp/platform.go

```diff
@@ -14,4 +14,9 @@ type Platform struct {
 	// platform configuration.
 	// +optional
 	DefaultMachinePlatform *MachinePool `json:"defaultMachinePlatform,omitempty"`
+
+	// VPC specifies a an existing cluster network where the cluster will be created
+	// rather than provisioning a new VPC
+	// +optional
+	VPC *VPC `json:"vpc,omitempty"`
```

Comment:
I would like to see these (network name, control plane subnetwork, compute subnetwork) fields in the platform structure rather than under new structure VPC.